cli_tools/common/utils/storage/scratch_bucket_creator.go

```diff
@@ -45,38 +46,50 @@ func NewScratchBucketCreator(ctx context.Context, storageClient domain.StorageCl
 // CreateScratchBucket creates scratch bucket in the same region as sourceFileFlag.
 // Returns (bucket_name, region, error)
 func (c *ScratchBucketCreator) CreateScratchBucket(
-	sourceFileFlag string, project string) (string, string, error) {
-	bucket := ""
-	region := ""
-	var err error
+	sourceFileFlag string, project string, zone string) (string, string, error) {
 
 	if project == "" {
 		return "", "", fmt.Errorf("can't create scratch bucket if project not specified")
 	}
+
 	if sourceFileFlag != "" {
 		// source file provided, create bucket in the same region for cost/performance reasons
-		bucket, region, err = c.createBucketMatchFileRegion(sourceFileFlag, project)
+		return c.createBucketMatchFileRegion(sourceFileFlag, project, zone)
 	}
-	if sourceFileFlag == "" || err != nil {
-		// source file not provided or couldn't create bucket based on it, create default scratch bucket
-		bucket = c.formatScratchBucketName(project, defaultRegion)
-		region, err = c.createBucketIfNotExisting(bucket, project, &storage.BucketAttrs{Name: bucket, Location: defaultRegion, StorageClass: defaultStorageClass})
-		if err != nil {
+
+	// if source file is not provided, fallback to input / default region.
+	return c.createBucketOnFallbackZone(project, zone)
+}
+
+func (c *ScratchBucketCreator) createBucketOnFallbackZone(project string, zone string) (string, string, error) {
```

Comment:
Move this method below createBucketMatchFile region as it's not the first to be called from CreateScratchBucket